enterprise/server/backends/distributed/distributed.go

```diff
@@ -1080,7 +1082,23 @@ func (c *Cache) Get(ctx context.Context, rn *rspb.ResourceName) ([]byte, error)
 		return nil, err
 	}
 	defer r.Close()
-	buf := new(bytes.Buffer)
+
+	var buf *bytes.Buffer
+	if rn.GetCacheType() == rspb.CacheType_CAS {
+		// If this is a CAS object, size the buffer to fit exactly.
+		buf = bytes.NewBuffer(make([]byte, 0, int(rn.GetDigest().GetSizeBytes())))
+	} else if strings.HasPrefix(rn.GetInstanceName(), content_addressable_storage_server.TreeCacheRemoteInstanceName) {
+		// If this is a TreeCache entry that we wrote; pull the size
+		// from the remote instance name.
+		parts := strings.Split(rn.GetInstanceName(), "/")
+		if s, err := strconv.Atoi(parts[len(parts)-1]); err == nil {
```

Comment:
this line looks like a well-crafted instance name could cause us to allocate an arbitrary-length array (if i'm understanding right).  could we clamp this value to something? (lmk if i'm misunderstanding)

Reply:
yeah that's a good point. I'll cap it to ~few MB

Reply:
Done.